packages/controller/main.js

```diff
@@ -2863,18 +2863,19 @@ async function processMessage(msg) {
 
         case 'writeDirAsZip':
             zipFiles = zipFiles || require('./lib/zipFiles');
-            try {
-                zipFiles.writeDirAsZip(
+            zipFiles
+                .writeDirAsZip(
                     objects,
                     msg.message.id,
                     msg.message.name,
                     Buffer.from(msg.message.data, 'base64'),
-                    msg.message.options,
-                    error => msg.callback && msg.from && sendTo(msg.from, msg.command, { error }, msg.callback)
-                );
-            } catch (error) {
-                msg.callback && msg.from && sendTo(msg.from, msg.command, { error }, msg.callback);
-            }
+                    msg.message.options
+                )
+                .then(() => msg.callback && msg.from && sendTo(msg.from, msg.command, {}, msg.callback))
```

Comment:
lets make the containing function async and use async/await

Reply:
Done